### PR TITLE
Release/1.8.0

### PR DESCRIPTION
**Pull request by b-butler**
## Description
Release minor version of signac.

### CODE REVIEW

**Comment by bdice** [comment]
I'd like to review/merge some form of this PR before releasing 1.8.0: https://github.com/glotzerlab/signac/pull/818

File: changelog.txt [hunk 1]
```diff
@@ -22,10 +22,12 @@ Changed
 +++++++
 
  - Schema migration is now performed on directories rather than signac projects and supports a wider range of schemas (#654).
- - Deprecated features now use ``FutureWarning`` instead of ``DeprecationWarning``, which is hidden by default (#687, #691).
+ - Deprecated features now use ``FutureWarning`` instead of ``DeprecationWarning``, which is hidden by default (#687, #691, #692).
  - Project names have a default in anticipation of removing names entirely. Project names will be removed in signac 2.0 (#644).
  - ``Project.workspace`` is now a property, not a method (#685).
  - Continuous integration uses GitHub Actions instead of CircleCI (#776, #788).
+ - Raise errors in testing when ``DeprecatedWarning``s or ``FutureWarning``s are raised (#713).
+ - Change GitHub PR to check for uncompleted tasks (i.e. unchecked checkboxes) (#686)
```
Comment (bdice) [inline changelog.txt:30]:
```suggestion
 - Change GitHub PR to check for uncompleted tasks (i.e. unchecked checkboxes) (#686).
```

**Comment by bdice** [comment]
@b-butler Can you help me understand what all the refactoring commits are doing? Are there warnings that you're trying to suppress? Are there tasks we missed that should be reviewed and merged into `master` as separate PRs? (Typically "release" PRs are trivial and only involve version bumps.)

**Comment by bdice** [comment]
I worked through the deprecations with @b-butler and @mikemhenry on #820 in real time.

**Comment by bdice** [comment]
Before this is ready to release, we need a new GitHub Action to deploy to the test PyPI server (and the real PyPI when we merge/tag).

**Comment by vyasr** [comment]
> Before this is ready to release, we need a new GitHub Action to deploy to the test PyPI server (and the real PyPI when we merge/tag).

@bdice @b-butler I handled this in #821.

**Comment by b-butler** [comment]
The package now builds wheels and publishes to TestPyPI correctly, so I think it should be good to merge.

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
Agreed! This is ready to go. @b-butler You can merge and finalize the GitHub release.

**Pull request merged**